include/swift/Option/FrontendOptions.td

```diff
@@ -321,6 +321,10 @@ def enable_experimental_async_top_level :
 def enable_experimental_bound_generic_extensions :
   Flag<["-"], "enable-experimental-bound-generic-extensions">,
   HelpText<"Enable experimental support for extensions of bound generic types">;
+
+def enable_experimental_ad_hoc_availability :
+  Flag<["-"], "enable-experimental-ad-hoc-availability">,
```

Comment:
From experience, I would advise to avoid using "experimental" in a flag name. A few of our experimental flags remain named as such even when used in normal builds today and it can make it harder for projects to adopt. If these are passed down by the driver changing it later can require a months long transition period. In general it's not worth it, I now prefer to instead hide flags for unsupported/experimental features.

Reply:
I had a hard time convincing some project owners to use the `-experimental-emit-module-separately` flag even when it was already the default on most branches and when their project didn't even build without it...

Reply:
Ok, that makes sense, thanks! I guess between hiding the flag and describing the experimental nature in the help string should probably be enough to communicate clearly whether its ready for adoption.